pubsub/benchmarks_test.go

```diff
@@ -0,0 +1,120 @@
+// Copyright 2019 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pubsub_test
+
+import (
+	"context"
+	"flag"
+	"testing"
+
+	"gocloud.dev/gcp"
+	"gocloud.dev/pubsub"
+	"gocloud.dev/pubsub/gcppubsub"
+	"golang.org/x/sync/errgroup"
+	"golang.org/x/sync/semaphore"
+)
+
+var (
+	projectID = flag.String("benchmark-project", "", "project ID")
```

Comment:
Can't we get the projectID from the creds?

https://github.com/google/go-cloud/blob/master/gcp/gcp.go#L87

Reply:
Doesn't work for me. My default creds come from ~/.config/gcloud/application_default_credentials.json, which doesn't have a project. @zombiezen any idea how to fix?

Reply:
I think you need to set it using `gcloud config set`:
https://cloud.google.com/sdk/gcloud/reference/config/set

Reply:
No, I have a project set in that way. Doesn't matter.

Reply:
@jba is right: explicit project ID is the right way to go here. IIRC there are examples of this around the Go Cloud repo.